samples/net/mqtt/src/modules/transport/transport.c

```diff
@@ -0,0 +1,344 @@
+{
+	ARG_UNUSED(return_code);
+
+	smf_set_state(SMF_CTX(&s_obj), &state[MQTT_CONNECTED]);
+}
+
+static void on_mqtt_disconnect(int result)
+{
+	ARG_UNUSED(result);
+
+	smf_set_state(SMF_CTX(&s_obj), &state[MQTT_DISCONNECTED]);
+}
+
+static void on_mqtt_publish(struct mqtt_helper_buf topic, struct mqtt_helper_buf payload)
+{
+	LOG_INF("Received payload: %.*s on topic: %.*s", payload.size,
+							 payload.ptr,
+							 topic.size,
+							 topic.ptr);
+}
+
+static void on_mqtt_suback(uint16_t message_id, int result)
+{
+	if ((message_id == SUBSCRIBE_TOPIC_ID) && (result == 0)) {
+		LOG_INF("Subscribed to topic %s", CONFIG_MQTT_SAMPLE_TRANSPORT_SUBSCRIBE_TOPIC);
+	}
+}
+
+/* Local convenience functions */
+static void publish(struct payload *payload)
+{
+	int err;
+	struct mqtt_publish_param param = {
+		.message.payload.data = payload->string,
+		.message.payload.len = strlen(payload->string),
+		.message.topic.qos = MQTT_QOS_1_AT_LEAST_ONCE,
+		.message_id = k_uptime_get_32(),
+		.message.topic.topic.utf8 = CONFIG_MQTT_SAMPLE_TRANSPORT_PUBLISH_TOPIC,
+		.message.topic.topic.size = strlen(CONFIG_MQTT_SAMPLE_TRANSPORT_PUBLISH_TOPIC),
+	};
+
+	err = mqtt_helper_publish(&param);
+	if (err) {
+		LOG_ERR("Failed to send payload, err: %d", err);
+		return;
+	}
+
+	LOG_INF("Publishing message: \"%.*s\" on topic: \"%.*s\"", param.message.payload.len,
+								   param.message.payload.data,
+								   param.message.topic.topic.size,
+								   param.message.topic.topic.utf8);
+}
+
+static void subscribe(void)
+{
+	int err;
+	struct mqtt_topic topics[] = {
+		{
+			.topic.utf8 = CONFIG_MQTT_SAMPLE_TRANSPORT_SUBSCRIBE_TOPIC,
+			.topic.size = strlen(CONFIG_MQTT_SAMPLE_TRANSPORT_SUBSCRIBE_TOPIC),
+		},
+	};
+	struct mqtt_subscription_list list = {
+		.list = topics,
+		.list_count = ARRAY_SIZE(topics),
+		.message_id = SUBSCRIBE_TOPIC_ID,
+	};
+
+	err = mqtt_helper_subscribe(&list);
+	if (err) {
+		LOG_ERR("Failed to subscribe to topics, error: %d", err);
+		return;
+	}
+}
+
+/* Connect work - Used to establish a connection to the MQTT broker and schedule reconnection
+ * attempts.
+ */
+static void connect_work_fn(struct k_work *work)
+{
+	ARG_UNUSED(work);
+
+	int err = client_id_get(client_id, sizeof(client_id));
+
+	if (err) {
+		LOG_ERR("client_id_get, error: %d", err);
+		return;
+	}
+
+	struct mqtt_helper_conn_params conn_params = {
+		.hostname.ptr = CONFIG_MQTT_SAMPLE_TRANSPORT_BROKER_HOSTNAME,
+		.hostname.size = strlen(CONFIG_MQTT_SAMPLE_TRANSPORT_BROKER_HOSTNAME),
+		.device_id.ptr = client_id,
+		.device_id.size = strlen(client_id),
+	};
+
+	err = mqtt_helper_connect(&conn_params);
+	if (err) {
+		LOG_ERR("Failed connecting to MQTT, error code: %d", err);
+	}
+
+	k_work_reschedule(&connect_work,
+			  K_SECONDS(CONFIG_MQTT_SAMPLE_TRANSPORT_RECONNECTION_TIMEOUT_SECONDS));
+}
+
+/* Zephyr State Machine framework handlers */
+static void disconnected_entry(void *o)
+{
+	struct s_object *od = o;
```

Comment:
It hink this function should have more descriptive variable names, `o` and `od` are hard to decipher. Same for other functions below too